Assets/Tests/InputSystem/Plugins/InputForUITests.cs

```diff
@@ -140,6 +144,37 @@ public void SendWheelEvent()
         Assert.That(m_InputForUIEvents[0].asPointerEvent.scroll, Is.EqualTo(new Vector2(0, 1)));
     }
 
+    [Test]
+    [Category("InputForUI")]
+    [TestCase(true)]
+    [TestCase(false)]
+    // The goal of this test is to make sure that InputSystemProvider works with and without project-wide actions asset
+    // so that there is no impact in receiving the necessary input events for UI.
+    // When there are no project-wide actions asset, the InputSystemProvider should still work as it currently gets
+    // the actions from DefaultActionsAsset().asset.
+    public void EventProviderWorksWithAndWithoutProjectWideActionsSet(bool useProjectWideActionsAsset)
+    {
+        Update();
+        if (!useProjectWideActionsAsset)
+        {
+            // Remove the project-wide actions asset in play mode and player.
+            // It will call InputSystem.onActionChange and re-set InputSystemProvider.actionAsset
+            // This the case where no project-wide actions asset is available in the project.
+            InputSystem.s_Manager.actions = null;
```

Comment:
I see this bypasses the play mode check by accessing internals directly, seems like a pragmatic solution that eliminates the need to mock via IInputRuntime interface discussed previously. Just maybe note that this is not representative of what happens in reality since this can only can happen in edit mode.

I assume the intention here is to test same behavior with and without PWA. However, this should be applicable for all tests in this class, and hence I would recommend either parameterising all test cases in this class or use two separate test classes that inherits the test methods and use different fixed setups instead?

Reply:
Also we need to not corrupt the state of `InputSystem.s_Manager.actions` after running this test. I would recommend either storing its value locally and setting it back in a `finally` block, or making the state be reset in `TearDown`.

Reply:
@benoitalain This was also my first thought, but I believe its not corrupted, its automatically reinitialised as part of InputTestFixture reinitialisation.